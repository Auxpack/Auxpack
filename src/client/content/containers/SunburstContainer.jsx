import React, { useState, useEffect } from 'react';
import Sunburst from '../../views/sunburst/Sunburst.jsx'
import { parse } from 'path';

const SunburstContainer = ({ build, activeBuild }) => {
    //console.log(`build props in SunburstContainer: `, build, activeBuild);

    const [defaultBar, setBar] = useState(false);
    const [dataBar, setDataBar] = useState([]);
    const [burstData, setData] = useState([]);
    const [activeBurst, setBurst] = useState(null);
    //console.log(`activeBuild `, activeBuild)

    // parse data, return array of arrays, each subarray contains [path, sizeString]
    const dataParser = () => {
        const data = build;

        //loops through assets
        let i = activeBuild;
        //console.log(`activeBuild in dataParser`, activeBuild)
        //let pathAry;
        let path;
        let sizeStr;
        let sunBurstData = [];
        // check if data is empty

        if (data.length !== 0) {
            for (let k = 0; k < data[i].chunks.length; k++) {
                for (let l = 0; l < data[i].chunks[k].modules.length; l++) {
                    sizeStr = data[i].chunks[k].modules[l].size.toString();
                    path = data[i].chunks[k].modules[l].name.replace("./", "");
                    sunBurstData.push([path, sizeStr])
                }
            }
            return sunBurstData;
        }
    } // end dataParser definition

    useEffect(() => {
        const parsedData = dataParser();
        setData(parsedData);
    }, [activeBuild]) // second arg: dependency that change

    const handleBurstHover = (path) => {
        setBurst(path)
    }


    return <React.Fragment>
        {(burstData !== undefined) ? <Sunburst
            burstData={burstData}
            onHover={handleBurstHover}
        /> : <h1>Loading...</h1>}
    </React.Fragment>
}

export default SunburstContainer;