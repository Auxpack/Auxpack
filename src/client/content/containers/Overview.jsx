import React from 'react'
import SunburstContainer from './SunburstContainer.jsx'

const Overview = ({ build, activeBuild }) => {
    //overview container for information if we wanted more than the sunburst on the first page
    return (
        <div id="container">
            <SunburstContainer build={build} activeBuild={activeBuild} />
        </div>
    );
}

export default Overview;