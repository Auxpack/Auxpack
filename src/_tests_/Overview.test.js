import Overview from '../client/content/containers/Overview.jsx';
import SunburstContainer from '../client/content/containers/SunburstContainer.jsx';

describe('Overview Unit Tests', () => {
  let wrapper;

  const props = {
    build: [{
      timeStamp: 1575426090404,
      time: 1439,
      hash: '546142ce1b49a6ba7d6f',
      errors: [],
      size: 1172113,
      assets: [{ name: 'bundle.js', chunks: ['main'], size: 1172113 }],
      chunks: [{ size: 1118609, files: ['bundle.js'], modules: [{ name: './client/App.jsx', size: 6375, id: './client/App.jsx' }] }],
      treeStats: { csj: [], esm: [], both: [] },
    }],
    activeBuild: 1,
  };

  beforeEach(() => {
    wrapper = shallow(<Overview {...props} />);
  });

  it('should render', () => {
    expect(wrapper);
  });

  test('Snapshot testing Overview component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Overview should render a div with id of 'container'", () => {
    expect(wrapper.find('#container').length).toEqual(1);
  });

  it('Overview should render a div wrapped around SunburstContainer', () => {
    const div = wrapper.find('#container');
    expect(div.find(SunburstContainer).length).toEqual(1);
  });
});
