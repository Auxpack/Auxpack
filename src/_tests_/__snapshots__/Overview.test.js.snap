// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Unit Tests Overview snapshot testing 1`] = `
<div
  id="container"
>
  <SunburstContainer
    activeBuild={1}
    build={
      Array [
        Object {
          "assets": Array [
            Object {
              "chunks": Array [
                "main",
              ],
              "name": "bundle.js",
              "size": 1172113,
            },
          ],
          "chunks": Array [
            Object {
              "files": Array [
                "bundle.js",
              ],
              "modules": Array [
                Object {
                  "id": "./client/App.jsx",
                  "name": "./client/App.jsx",
                  "size": 6375,
                },
              ],
              "size": 1118609,
            },
          ],
          "errors": Array [],
          "hash": "546142ce1b49a6ba7d6f",
          "size": 1172113,
          "time": 1439,
          "timeStamp": 1575426090404,
          "treeStats": Object {
            "both": Array [],
            "csj": Array [],
            "esm": Array [],
          },
        },
      ]
    }
  />
</div>
`;
